packages/core/nano/src/core.ts

```diff
@@ -175,5 +175,4 @@ export const ops: OpContext = {
     negate: liftUnaryOp((x: any) => -x),
 };
 
-export type Formula = <O>(host: O, context: CalcValue<O>) => [Pending<unknown>[], Delayed<CalcValue<O>>];
-
+export type Formula = <T = unknown, O = unknown>(host: O, context: CalcValue<T>) => [Pending<unknown>[], Delayed<CalcValue<T>>];
```

Comment:
The host (that should probably be called origin) only exists to be passed to CalcObj's in their `read` method so that they can appropriately track where the read has come from. If we make `T` and `O` distinct then there is no meaningful way to use the origin.